isis/tests/TgoCassisModuleTests.cpp

```diff
@@ -1146,3 +1150,389 @@ TEST(TgoCassisModuleTests, TgoCassisIngestReingest) {
+  UserInterface tgocassismosBlu(MOS_XML, cassismosArgs);
+  try {
+    tgocassismos(tgocassismosBlu);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to run tgocassismos on blu image: " << e.what() << std::endl;
+  }
+
+  // run cam2map and cassismos on red cube
+  QString redEquiFile = prefix.path() + "/red_equi.cub";
+  cam2mapArgs = {"from=" + redFileName,
+                  "to=" + redEquiFile,
+                  "map=" + mapFile,
+                  "defaultrange=map",
+                  "pixres=mpp",
+                  "resolution=200"};
+  UserInterface cam2mapRed(CAM2MAP_XML, cam2mapArgs);
+  try {
+    cam2map(cam2mapRed);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to run cam2map on red image: " << e.what() << std::endl;
+  }
+
+  FileList *redMosaicList = new FileList();
+  redMosaicList->append(redEquiFile);
+  QString redListFile = prefix.path() + "/redMosaic.lis";
+  redMosaicList->write(redListFile);
+
+  QString redCassisMosaic = prefix.path() + "/redCassisMosaic.cub";
+  cassismosArgs = {"fromlist=" + redListFile, "to=" + redCassisMosaic};
+  UserInterface tgocassismosRed(MOS_XML, cassismosArgs);
+  try {
+    tgocassismos(tgocassismosRed);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to run tgocassismos on red image: " << e.what() << std::endl;
+  }
+
+
+  // run cubeit
+  FileList *MosaicList = new FileList();
+  MosaicList->append(redCassisMosaic);
+  MosaicList->append(bluCassisMosaic);
+  MosaicList->append(nirCassisMosaic);
+  MosaicList->append(panCassisMosaic);
+  QString mosListFile = prefix.path() + "/mosaicList.lis";
+  MosaicList->write(mosListFile);
+
+  QString coloredMosaic = prefix.path() + "/coloredMosaic.cub";
+  QVector<QString> cubeitArgs = {"fromlist=" + mosListFile, "to=" + coloredMosaic};
+  UserInterface cubeitUI(CUBEIT_XML, cubeitArgs);
+  try {
+    cubeit(cubeitUI);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to run cubeit on mosaic list: " << e.what() << std::endl;
+  }
+
+  // Mosaic Cube
+  Cube mosCube(coloredMosaic);
+  Pvl *lab = mosCube.label();
+
+  // Instrument Group
+  PvlGroup mos = lab->findGroup("Mosaic", Pvl::Traverse);
+  EXPECT_EQ(mos["SpacecraftName"][0].toStdString(), "TRACE GAS ORBITER");
+  EXPECT_EQ(mos["InstrumentId"][0].toStdString(), "CaSSIS");
+  EXPECT_EQ(mos["ObservationId"][0].toStdString(), "CRUS_049218_201_0" );
+  EXPECT_EQ(mos["StartTime"][0].toStdString(), "2016-11-26T22:50:27.381");
+  EXPECT_EQ(mos["StopTime"][0].toStdString(), "2016-11-26T22:50:27.382");
+  EXPECT_EQ(mos["SpacecraftClockStartCount"][0].toStdString(), "2f015435767e275a");
+  EXPECT_DOUBLE_EQ(double(mos["IncidenceAngle"]), 44.946650468616);
+  EXPECT_DOUBLE_EQ(double(mos["EmissionAngle"]), 11.637754697441);
+  EXPECT_DOUBLE_EQ(double(mos["LocalTime"]), 14.429448515306);
+  EXPECT_DOUBLE_EQ(double(mos["SolarLongitude"]), 269.1366003982);
+  EXPECT_DOUBLE_EQ(double(mos["SubSolarAzimuth"]), 139.56469945225);
+  EXPECT_DOUBLE_EQ(double(mos["NorthAzimuth"]), 270.0);
```

Comment:
I know this was a fair amount of copy/paste and small edits but we should change it to use what the `functionaltestsciss2isis` are using. See [here](https://github.com/USGS-Astrogeology/ISIS3/blob/5b23dce4ba48dbdefaed8d6c0680fd3bd55aff2d/isis/tests/FunctionalTestsCiss2isis.cpp#L44) from line 44 to 80. Basically copy the expected group output, then do a direct comparison with that group and the group that is produced on the mosaic label